homeassistant/components/tado/climate.py

```diff
@@ -496,6 +506,12 @@ def set_hvac_mode(self, hvac_mode: HVACMode) -> None:
         """Set new target hvac mode."""
         self._control_hvac(hvac_mode=HA_TO_TADO_HVAC_MODE_MAP[hvac_mode])
 
+    def set_heating_circuit(self, heater=None):
+        #TODO: Convert heater entity ID to water heater SerialNo and find its ID by self._tado.circuits
```

Comment:
It would be necessary to create a fully new entity for each home water heater, especially in cases where the water heater (Wolf CGB(-K)-20) has separate circuits for heating the house and providing hot water. Currently, the code is designed to select from a water_heater entity that is specifically for hot water, not for heating the house. While this is primarily a frontend issue and doesn't affect the underlying functionality, it's still important to address. Additionally, the current logic only supports basic on/off control for heat zones, lacking the ability to select specific heating circuit.

Reply:
ok, I don't think I understand, sorry. Could you please give a bit more context about what this service is supposed to do and how it's supposed to work?

Reply:
https://github.com/user-attachments/assets/aff188d8-679d-427a-acb6-05138d0c055d

It basically does this thing, but via API. As shown in video.